buildkite/pipelines.go

```diff
@@ -232,25 +236,24 @@ func (ps *PipelinesService) Delete(org string, slug string) (*Response, error) {
 // Update - Updates a pipeline.
 //
 // buildkite API docs: https://buildkite.com/docs/rest-api/pipelines#update-a-pipeline
-func (ps *PipelinesService) Update(org, slug string, p *UpdatePipeline) (*Pipeline, *Response, error) {
+func (ps *PipelinesService) Update(org string, p *UpdatePipeline) (*Response, error) {
```

Comment:
Oh wait but this is still a breaking change isn't it? (compared to the last release at least).

I thought you mean something more like this:

```go
func (ps *PipelinesService) Update(org string, p *Pipeline) (*Response, error) {
   update := UpdatePipeline{
      Name: p.Name,
      ...
   }
...
```

So that the function footprint is compatible with the current release, but internal we start moving it towards an UpdatePipeline function argument. And we can look to make a breaking change in v4 and consolidate Update and Create so they work similarly.

UpdatePipeline struct can define all the omitempty properties so that it only marshals to json values that have been set to non-default values. But this may cause a problem for any pipeline settings that the user wants to specifically set to false, which is where the map[string]interface approach would work for I think

Reply:
I agree with @jradtilbrook  on this one. This is still a breaking change.  We can start applying such when we plan to do a v4 release.

Reply:
Re-implemented with the above in mind 👍